### DISCUSSION TITLE
kafka streams javaagent

### DISCUSSION THREAD

**Discussion by dermotmburke** [open-telemetry/opentelemetry-java-instrumentation#7620] [Q&A] [answered]
Hi,

I came across https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/kafka/kafka-streams-0.11/javaagent but am unsure how tom use it. I tried downloading the jar file from here https://search.maven.org/artifact/io.opentelemetry.javaagent.instrumentation/opentelemetry-javaagent-kafka-streams-0.11 but when I try to use it in a `java -javaagent:opentelemetry-javaagent-kafka-streams-0.11-1.22.1-alpha.jar -jar app.jar` command I get the following error

```
Failed to find Premain-Class manifest attribute in opentelemetry-javaagent-kafka-streams-0.11-1.22.1-alpha.jar
Error occurred during initialization of VM
agent library failed to init: instrument
```

Am I missing something?

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @dermotmburke ,

That library is just a part of the OTel javaagent (one of many), you can't really do much with it alone (we release it for people that want to extend it or develop a custom javaagent instrumentation that reference it). You should be using the OTel javaagent instead.
Try the following instructions: https://opentelemetry.io/docs/instrumentation/java/automatic/

**Reply by dermotmburke** [reply]
ahh ok - I have tried the otel javaagent but thought this would give me more specific metrics for kafka streams applications. Thanks for getting back to me.
